gr-qtgui/python/qtgui/ledindicator.py

```diff
@@ -0,0 +1,207 @@
+
+from gnuradio import gr
+import pmt
+
+class LabeledLEDIndicator(QFrame):
+    # Positions: 1 = above, 2=below, 3=left, 4=right
+    def __init__(self, lbl='', onColor='green', offColor='red', initialState=False, maxSize=80, position=1, alignment=1, valignment=1, parent=None):
+        QFrame.__init__(self, parent)
+        self.numberControl = LEDIndicator(onColor, offColor, initialState, maxSize, parent)
+        
+        if position < 3:
+            layout =  QVBoxLayout()
+        else:
+            layout = QHBoxLayout()
+            
+        if len(lbl) == 0:
+            lbl = " "
+        self.lbl = lbl
+        self.lblcontrol = QLabel(lbl, self)
+        self.lblcontrol.setAlignment(Qtc.AlignCenter)
+
+        # add top or left
+        if len(lbl) > 0:
+            if position == 1 or position == 3:
+                layout.addWidget(self.lblcontrol)
+        else:
+            self.hasLabel = False
+
+        layout.addWidget(self.numberControl)
+        
+        # Add bottom or right
+        if len(lbl) > 0:
+            if position == 2 or position == 4:
+                layout.addWidget(self.lblcontrol)
+                
+        if alignment == 1:        
+            halign = Qtc.AlignCenter
+        elif alignment == 2:
+            halign = Qtc.AlignLeft
+        else:
+            halign = Qtc.AlignRight
+
+        if valignment == 1:
+            valign = Qtc.AlignVCenter
+        elif valignment == 2:
+            valign = Qtc.AlignTop
+        else:
+            valign = Qtc.AlignBottom
+            
+        layout.setAlignment(halign | valign)
+        self.setLayout(layout)
+        
+        if (len(lbl) > 0):
+            textfont = self.lblcontrol.font()
+            metrics = QFontMetricsF(textfont)
+            
+            maxWidth = max( (maxSize+30),(maxSize + metrics.width(lbl)+4) )
+            maxHeight = max( (maxSize+35),(maxSize + metrics.height()+2) )
+            self.setMinimumSize(maxWidth, maxHeight)
+        else:
+            self.setMinimumSize(maxSize+2, maxSize+2)  
+
+        self.show()
+        
+    def setState(self,onOff):
+        self.numberControl.setState(onOff)
+        
+class LEDIndicator(QFrame):
+    def __init__(self, onColor='green', offColor='red', initialState=False, maxSize=80, parent=None):
+        QFrame.__init__(self, parent)
+
+        #super().setPixmap(QtGui.QPixmap(":/icons/led-red-on.png"))
+        #super().setScaledContents(True)
+
+        self.maxSize = maxSize
+        self.curState = initialState  
+        self.onColor = QColor(onColor)
+        self.offColor = QColor(offColor)
+
+        self.setMinimumSize(maxSize, maxSize)      
+        self.setMaximumSize(maxSize, maxSize)  
+
+    def setState(self,onOff):
+        self.curState = onOff
+        super().update()
+        
+    def paintEvent(self, event):
+        super().paintEvent(event)
+        
+        painter = QPainter(self)
+        
+        #self.painter.setPen(self.backgroundColor)
```

Comment:
should all the commented out lines here be rm'ed? not going to highlight all of the but between here and ~176 there is several